doc/source/whatsnew/v0.23.0.txt

```diff
@@ -864,6 +864,7 @@ I/O
 - Bug in :func:`read_csv` where missing values were not being handled properly when ``keep_default_na=False`` with dictionary ``na_values`` (:issue:`19227`)
 - Bug in :func:`read_sas` where a file with 0 variables gave an ``AttributeError`` incorrectly. Now it gives an ``EmptyDataError`` (:issue:`18184`)
 - Bug in :func:`DataFrame.to_latex()` where pairs of braces meant to serve as invisible placeholders were escaped (:issue:`18667`)
+- Bug in :func:`DataFrame.to_latex()` where a ``NaN`` in a ``MultiIndex`` would cause an ``IndexError`` or incorrect output (:issue:`14249`)
```

Comment:
where an all-``NaN`` level in a ``MultiIndex``

(I think)

Reply:
In the SO link in #14249 there is a case where `NaN`s are being silently replaced with other non-`NaN` values in the level and this fixes both that issue and the one actually reported

Reply:
right!